Frameworks/CoreGraphics/D2DWrapper.mm

```diff
@@ -0,0 +1,33 @@
+//******************************************************************************
+//
+// Copyright (c) Microsoft. All rights reserved.
+//
+// This code is licensed under the MIT License (MIT).
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+
+#import <CoreGraphics/D2DWrapper.h>
+#import <Starboard.h>
+
+using namespace Microsoft::WRL;
+
+// Private helper for creating a D2DFactory
+static ComPtr<ID2D1Factory> __CreateD2DFactoryInstance() {
+    ComPtr<ID2D1Factory> d2dFactory;
+    THROW_IF_FAILED(D2D1CreateFactory(D2D1_FACTORY_TYPE_MULTI_THREADED, __uuidof(ID2D1Factory), &d2dFactory));
```

Comment:
Any code that needs the multi-threaded access should lock what it needs.  making all D2D methods thread safe is not the way to do that.  Note that singlethreaded factory doesnt mean the objects cant be used in multiple threads - it just means that the caller has to ensure thread safety.  

---
In reply to: [82073552](https://github.com/Microsoft/WinObjC/pull/1085#discussion_r82073552) [](ancestors = 82073552)